R/as-data-frame.R

```diff
@@ -73,3 +73,54 @@ as.data.frame.CrunchDataFrame <- function (x, row.names = NULL, optional = FALSE
     names(out) <- names(x)
     return(structure(out, class="data.frame", row.names=c(NA, -nrow(x))))
 }
+
+
+#' as.data.frame method for VariableCatalog
+#'
+#' This method allows you to import the VariableCatalog into your R sessions in 
+#' order to facilliate interactive use. Modifying the dataframe produced by this
+#' method will not update the Crunch web app, for information about modifying 
+#' variables in the app see: \code{vingette("variables",
+#' package = "crunch")}  
+#' 
+#' @param varCat
+#' A VariableCatalog produced by \code{variables(ds)}.
+#' @param fields 
+#' A character vector of the fields from the variable catalog which you would like 
+#' included in the datafame. To include all fields or see which fields are available
+#' set fields to "all". 
+#' @return A data frame
+#' @examples 
+#' \dontrun{
+#' ds <- loadDataset("iris")
+#' vars <- variables(ds)
+#' var_df <- as.data.frame(vars, fields = "all")
+#' }
+#' 
+#' @rdname VariableCatalog-to-Data-Frame
+#' @importFrom purrr map_df map_lgl
+#' @importFrom dplyr bind_cols
+#' @export
+as.data.frame.VariableCatalog <- function(varCat, 
+                                          fields = c("alias", "name", "type")) {
+  out <- catalogToDataFrame(varCat)
+  subvariable_names <- c("subvariables", "subvariables_catalog")
+  sub_variables <- out[, subvariable_names]
+  vectors <- out[, setdiff(names(out), subvariable_names)]
+  ## Null values in the description field cause the description column to be
+  ## shorter than the other columns. Recoding as a character value makes the data
+  ## rectangular. 
+  vectors$description[map_lgl(vectors$description, is.null)] <-  "NULL"
+  vectors <- map_df(vectors, unlist)
+  out <- bind_cols(vectors, sub_variables)
+
+ if (all(fields == "all")) {
+   return(out)
+ }
+ if (!all(fields %in% names(out))) {
+   stop("Field name not present in variable catalog, use fields = 'all' to see available fields.")
```

Comment:
`stop` was my first inclination as well, but we have a `halt` function from `httpcache` that we've been using. `halt` adds some logging as well as by default hides the call from users.